tests/compilation/test_preprocessing.py

```diff
@@ -174,3 +174,45 @@ def test_precompile_propagates_linked_params(input_dict, expected_linked_params,
             routine = routine.children[part]
 
         assert routine.linked_params == linked_params
+
+
+def test_propagating_resources_works_as_expected_in_presence_of_large_number_of_children(backend):
```

Comment:
How about using this: https://pypi.org/project/pytest-timeout/ ?

Reply:
As discussed, we decided to give pytest-timeout a go, and it was incorporated into this PR.